src/slam_toolbox_common.cpp

```diff
@@ -522,7 +523,10 @@ bool SlamToolbox::shouldProcessScan(
 
   // check moved enough, within 10% for correction error
   const double dist2 = last_pose.SquaredDistance(pose);
-  if (dist2 < 0.8 * min_dist2 || scan_ctr < 5) {
+  double heading_diff = pose.GetHeading() - last_pose.GetHeading();
+  if (heading_diff < 0)
+      heading_diff = -heading_diff;
```

Comment:
You can just use `fabs()` in the `if` condition below and remove this

Reply:
OK.